### ISSUE TITLE
triggering focus event with namespace runs all focus listeners

### ISSUE THREAD

**Issue by mgol** [jquery/jquery#1750]
Originally reported by peter@… at: http://bugs.jquery.com/ticket/14359

<p>
Using jQuery 1.9 onwards, calling .trigger('focus.myevent') focusses the target element and triggers all "focus" handlers, regardless of namespace.
</p>

<p>
This is inconsistent with other events, where only handlers with the "myevent" namespace would be triggered.
</p>

<p>
<a class="ext-link" href="http://jsfiddle.net/PeterChaplin/GUSN5/" rel="nofollow"><span class="icon">&nbsp;</span>http://jsfiddle.net/PeterChaplin/GUSN5/</a>
</p>

<p>
This behaviour does not appear to be present in 1.7.2 or 1.8.3.
</p>

<p>
I have only tested this on Google Chrome Version 29.0.1547.66 m, in Windows 7
</p><p>Issue reported for jQuery 1.10.2</p>

**Comment by timmywil** [comment]
Related: https://github.com/jquery/jquery/issues/1735

**Comment by caseyjhol** [comment]
This might be an acceptable workaround to prevent firing the handler twice as mentioned in https://github.com/jquery/jquery/issues/1741#issuecomment-59859973:

```js
var $x = $("#aninput");
$x.triggerHandler("focus.myevent", ['extra', 'data']);
$x[0].focus();

$x.on("focus.myevent", function (e, extra, data) {
    if (e.originalEvent) return;
    console.log(extra, data);
});

$x.on("focus.notfired", function (e, extra, data) {
    if (e.originalEvent) return;
    console.log("This won't be fired.");
});
```

**Comment by gibson042** [comment]
Fixed by 669f720edc4f557dfef986db747c09ebfaa16ef5: http://jsfiddle.net/3nwhq0o7/1/
